ui/app/AppLayouts/Profile/popups/AddSocialLinkModal.qml

```diff
@@ -0,0 +1,126 @@
+        readonly property int selectedLinkType: d.selectedLinkIndex !== -1 ? staticLinkTypesModel.get(d.selectedLinkIndex).type : 0
+        readonly property string selectedLinkTypeText: d.selectedLinkIndex !== -1 ? staticLinkTypesModel.get(d.selectedLinkIndex).text : ""
+        readonly property string selectedIcon: d.selectedLinkIndex !== -1 ? staticLinkTypesModel.get(d.selectedLinkIndex).icon : ""
+
+        readonly property var staticLinkTypesModel: ListModel {
+            readonly property var data: [
+                { type: Constants.socialLinkType.twitter, icon: "twitter", text: "__twitter" },
+                { type: Constants.socialLinkType.personalSite, icon: "language", text: "__personal_site" },
+                { type: Constants.socialLinkType.github, icon: "github", text: "__github" },
+                { type: Constants.socialLinkType.youtube, icon: "youtube", text: "__youtube" },
+                { type: Constants.socialLinkType.discord, icon: "discord", text: "__discord" },
+                { type: Constants.socialLinkType.telegram, icon: "telegram", text: "__telegram" },
+                { type: Constants.socialLinkType.custom, icon: "link", text: "" }
+            ]
+            Component.onCompleted: append(data)
+        }
+    }
+
+    onCurrentIndexChanged: {
+        //StatusAnimatedStack doesn't handle well items' visibility,
+        //keeping this solution for now until #8024 is fixed
+        if (currentIndex === 1) {
+            customTitle.input.edit.clear()
+            linkTarget.input.edit.clear()
+            if (d.selectedLinkType === Constants.socialLinkType.custom)
+                customTitle.input.edit.forceActiveFocus()
+            else
+                linkTarget.input.edit.forceActiveFocus()
+        }
+    }
+
+    stackItems: [
+        StatusListView {
+            width: root.availableWidth
+            height: root.availableHeight
+            model: d.staticLinkTypesModel
+            delegate: StatusListItem {
+                width: ListView.view.width
+                title: ProfileUtils.linkTypeToText(model.type) || qsTr("Custom link")
+                asset.name: model.icon
+                asset.color: ProfileUtils.linkTypeColor(model.type)
+                onClicked: {
+                    d.selectedLinkIndex = index
+                    root.currentIndex++
+                }
+                components: [
+                    StatusIcon {
+                        anchors.verticalCenter: parent.verticalCenter
+                        icon: "next"
+                        color: Theme.palette.baseColor1
+                    }
+                ]
+            }
+        },
+        ColumnLayout {
+            width: root.availableWidth
+            spacing: Style.current.halfPadding
+
+            StaticSocialLinkInput {
+                id: customTitle
+                Layout.fillWidth: true
+                visible: d.selectedLinkType === Constants.socialLinkType.custom
```

Comment:
Yup, I'll add the character limit to QML, it's set to `24` in status-go. It's true that it might silently fail saving but the only validation status-go does is checking for non-empty URL (we don't allow that in QML) and the above length (will add)

Reply:
Added `Constants.maxSocialLinkTextLength: 24` and using it for the social link text fields (that's what the status-go validation does):

```go
const (
    maxSocialLinkTextLength = 24
)

func ValidateSocialLinks(socialLinks *identity.SocialLinks) error {
    for _, link := range *socialLinks {
        if len(link.Text) > maxSocialLinkTextLength {
            return ErrInvalidSocialLinkTextLength
        }
    }
    return nil
}
```

Reply:
![image](https://user-images.githubusercontent.com/5377645/225467368-8594cedc-8efa-43ae-b0ce-4c3de27bf129.png)